test/common/ratelimit/ratelimit_impl_test.cc

```diff
@@ -30,95 +30,116 @@ class MockRequestCallbacks : public RequestCallbacks {
     Http::HeaderMapImpl headers;
     GrpcClientImpl::createRequest(request, "foo", {{{{"foo", "bar"}, {"bar", "baz"}}}});
     EXPECT_CALL(*async_client_, send(_, ProtoEq(request), _, _, _))
         .WillOnce(Return(&async_request_));
 
-    client_.limit(request_callbacks_, "foo", {{{{"foo", "bar"}, {"bar", "baz"}}}},
-                  Tracing::NullSpan::instance());
+    client_->limit(request_callbacks_, "foo", {{{{"foo", "bar"}, {"bar", "baz"}}}},
+                   Tracing::NullSpan::instance());
 
-    client_.onCreateInitialMetadata(headers);
+    client_->onCreateInitialMetadata(headers);
 
-    response.reset(new pb::lyft::ratelimit::RateLimitResponse());
-    response->set_overall_code(pb::lyft::ratelimit::RateLimitResponse_Code_OK);
+    response.reset(new envoy::service::ratelimit::v2::RateLimitResponse());
+    response->set_overall_code(envoy::service::ratelimit::v2::RateLimitResponse_Code_OK);
     EXPECT_CALL(span_, setTag("ratelimit_status", "ok"));
     EXPECT_CALL(request_callbacks_, complete(LimitStatus::OK));
-    client_.onSuccess(std::move(response), span_);
+    client_->onSuccess(std::move(response), span_);
   }
 
   {
-    pb::lyft::ratelimit::RateLimitRequest request;
+    envoy::service::ratelimit::v2::RateLimitRequest request;
     GrpcClientImpl::createRequest(
         request, "foo",
         {{{{"foo", "bar"}, {"bar", "baz"}}}, {{{"foo2", "bar2"}, {"bar2", "baz2"}}}});
     EXPECT_CALL(*async_client_, send(_, ProtoEq(request), _, _, _))
         .WillOnce(Return(&async_request_));
 
-    client_.limit(request_callbacks_, "foo",
-                  {{{{"foo", "bar"}, {"bar", "baz"}}}, {{{"foo2", "bar2"}, {"bar2", "baz2"}}}},
-                  Tracing::NullSpan::instance());
+    client_->limit(request_callbacks_, "foo",
+                   {{{{"foo", "bar"}, {"bar", "baz"}}}, {{{"foo2", "bar2"}, {"bar2", "baz2"}}}},
+                   Tracing::NullSpan::instance());
 
-    response.reset(new pb::lyft::ratelimit::RateLimitResponse());
+    response.reset(new envoy::service::ratelimit::v2::RateLimitResponse());
     EXPECT_CALL(request_callbacks_, complete(LimitStatus::Error));
-    client_.onFailure(Grpc::Status::Unknown, "", span_);
+    client_->onFailure(Grpc::Status::Unknown, "", span_);
   }
 }
 
-TEST_F(RateLimitGrpcClientTest, Cancel) {
-  std::unique_ptr<pb::lyft::ratelimit::RateLimitResponse> response;
+TEST_P(RateLimitGrpcClientTest, Cancel) {
+  setClient(GetParam());
+  std::unique_ptr<envoy::service::ratelimit::v2::RateLimitResponse> response;
 
   EXPECT_CALL(*async_client_, send(_, _, _, _, _)).WillOnce(Return(&async_request_));
 
-  client_.limit(request_callbacks_, "foo", {{{{"foo", "bar"}}}}, Tracing::NullSpan::instance());
+  client_->limit(request_callbacks_, "foo", {{{{"foo", "bar"}}}}, Tracing::NullSpan::instance());
 
   EXPECT_CALL(async_request_, cancel());
-  client_.cancel();
+  client_->cancel();
 }
 
 TEST(RateLimitGrpcFactoryTest, Create) {
```

Comment:
I was thinking about explicitly testing the factory to make sure the correct Impl is being created (rather than the implicit testing done above), but I don't have a good sense on how to accomplish that other than type checking. Thoughts?

Reply:
I think the implicit testing is probably fine as long as we cover both cases for now.